ui/src/app/cohort-search/tree-node/tree-node.component.tsx

```diff
@@ -0,0 +1,347 @@
+    const {node: {code, conceptId, domainId, id}} = this.props;
+    return `param${!!conceptId && domainId !== DomainType.SURVEY.toString() ? (conceptId + code) : id}`;
+  }
+
+  get isPMCat() {
+    return [CriteriaSubType.WHEEL, CriteriaSubType.PREG, CriteriaSubType.HRIRR, CriteriaSubType.HRNOIRR]
+      .map(st => st.toString())
+      .includes(this.props.node.subtype);
+  }
+
+  get showCount() {
+    const {node: {code, count, group, selectable, subtype, type}} = this.props;
+    return count > -1 &&
+      (selectable || (subtype === CriteriaSubType.LAB.toString() && group && code !== null) || type === CriteriaType.CPT4.toString());
+  }
+
+  select(event: Event) {
+    event.stopPropagation();
+    const {node, node: {conceptId, domainId, group, parentId, subtype, value}, select, selectedIds} = this.props;
+    let {node: {name}} = this.props;
+    if (!selectedIds.includes(this.paramId)) {
+      let attributes = [];
+      if (subtype === CriteriaSubType.BP.toString()) {
+        Object.keys(PREDEFINED_ATTRIBUTES).forEach(key => {
+          if (name.indexOf(key) === 0) {
+            attributes = PREDEFINED_ATTRIBUTES[key];
+          }
+        });
+      } else if (this.isPMCat) {
+        attributes.push({
+          name: AttrName.CAT,
+          operator: Operator.IN,
+          operands: [value]
+        });
+      } else if (domainId === DomainType.SURVEY.toString() && !group) {
+        const question = ppiQuestions.getValue()[parentId];
+        if (question) {
+          name = `${question.name} - ${name}`;
+        }
+        const attribute = conceptId === 1585747
+          ? {name: AttrName.NUM, operator: Operator.EQUAL, operands: [value]}
+          : {name: AttrName.CAT, operator: Operator.IN, operands: [value]};
+        attributes.push(attribute);
+      }
+      const param = {
+        ...node as Object,
+        parameterId: this.paramId,
+        attributes,
+        name
+      };
+      select(param);
+    }
+  }
+
+  render() {
+    const {autocompleteSelection, groupSelections, node,
+      node: {code, count, domainId, id, group, hasAttributes, name, parentId, selectable}, scrollToMatch, searchTerms, select, selectedIds,
+      setAttributes} = this.props;
+    const {children, error, expanded, hover, loading, searchMatch} = this.state;
+    const nodeChildren = domainId === DomainType.PHYSICALMEASUREMENT.toString() ? node.children : children;
+    const selected = selectedIds.includes(this.paramId) || groupSelections.includes(parentId);
+    const displayName = domainId === DomainType.PHYSICALMEASUREMENT.toString() && !!searchTerms
+      ? highlightSearchTerm(searchTerms, name, colors.success)
+      : name;
+    return <React.Fragment>
+      <div style={{...styles.treeNode}} id={`node${id}`} onClick={() => this.toggleExpanded()}>
+        {group && <button style={styles.iconButton}>
+          {loading
+            ? <Spinner size={16}/>
+            : <ClrIcon style={{color: colors.disabled}}
```

Comment:
Opt Nit: Should this be a FontAwesome icon?

Reply:
I believe the plan is for us to fully move over to FA. Might be worth checking with Lou, but happy with this being a follow up.